src/install/install.zig

```diff
@@ -2815,9 +2834,20 @@ pub const PackageManager = struct {
                             const res_tag = resolutions[list.items[0]].tag;
                             const ver_tag = version.tag;
                             if ((res_tag == .npm and ver_tag == .npm) or (res_tag == .git and ver_tag == .git) or (res_tag == .github and ver_tag == .github)) {
+                                const existing_package = this.lockfile.packages.get(list.items[0]);
+                                this.log.addWarningFmt(
+                                    null,
+                                    logger.Loc.Empty,
+                                    this.allocator,
+                                    "incorrect peer dependency \"{}@{}\"",
+                                    .{
+                                        existing_package.name.fmt(this.lockfile.buffers.string_bytes.items),
+                                        existing_package.resolution.fmt(this.lockfile.buffers.string_bytes.items),
+                                    },
+                                ) catch unreachable;
                                 successFn(this, dependency_id, list.items[0]);
                                 return .{
-                                    .package = this.lockfile.packages.get(list.items[0]),
+                                    .package = existing_package,
```

Comment:
// we must fetch it from the packages array again, incase the package array mutates the value in the `successFn`
                                    .package = this.lockfile.packages.get(existing_package_id),